src/index.ts

```diff
@@ -67,42 +67,54 @@ type Tag<A extends AnyMember> = A[TagKey]
 type Value<A extends AnyMember> = A[ValueKey]
 
 /**
- * A type-level representation of the `mkConstructor` function.
+ * A constructor is either `A -> B` or, if it's nullary, directly `B`.
```

Comment:
Should we document somewhere why we decided to do this? Mainly for people outside of Unsplash who might consider using this library.